src/dir_contents.ml

```diff
@@ -426,23 +448,197 @@ let build_mlds_map (d : Super_context.Dir_with_jbuild.t) ~files =
       Some (doc, List.map (String.Map.values mlds) ~f:(Path.relative dir))
     | _ -> None)
 
-let get =
-  let cache = Hashtbl.create 32 in
-  fun sctx ~dir ->
-    Hashtbl.find_or_add cache dir ~f:(fun dir ->
-      match Super_context.stanzas_in sctx ~dir with
-      | None ->
-        { text_files = String.Set.empty
-        ; modules = lazy
-            { libraries = String.Map.empty
-            ; executables = String.Map.empty
-            ; rev_map = Module.Name.Map.empty
-            }
+module Dir_status = struct
+  type t =
+    | Empty_standalone of File_tree.Dir.t option
+    (* Directory with no libraries or executables that is not part of
+       a multi-directory group *)
+
+    | Is_component_of_a_group_but_not_the_root of
+        Super_context.Dir_with_jbuild.t option
+    (* Sub-directory of a directory with [(include_subdirs true)] *)
```

Comment:
This commend needs to be updated as you've changed the argument for this stanza.